osu.Game/Database/BeatmapInfo.cs

```diff
@@ -13,11 +13,15 @@ namespace osu.Game.Database
 {
     public class BeatmapInfo : IEquatable<BeatmapInfo>
     {
-        [PrimaryKey]
-        public int BeatmapID { get; set; }
+        [PrimaryKey, AutoIncrement]
+        public int ID { get; set; }
+
+        public int BeatmapID { get; set; } = 0;
+
+        public int BeatmapSetID { get; set; } = -1;

```

Comment:
please make this nullable with null default, and maybe rename to OnlineBeatmapSetID